internal/hwmon/hwmon.go

```diff
@@ -292,3 +280,39 @@ func findPlatform(devicePath string) string {
 	platformRegex := regexp.MustCompile(".*/platform/{}/.*")
 	return platformRegex.FindString(devicePath)
 }
+
+func UpdateFanConfigFromHwMonControllers(controllers []*HwMonController, config *configuration.FanConfig) error {
+	for _, controller := range controllers {
+		matched, err := regexp.MatchString("(?i)"+config.HwMon.Platform, controller.Platform)
+		if err != nil {
+			return errors.New(fmt.Sprintf("Failed to match platform regex of %s (%s) against controller platform %s", config.ID, config.HwMon.Platform, controller.Platform))
+		}
+		if !matched {
+			continue
+		}
+		for _, fan := range controller.Fans {
+			controllerConfig := fan.Config.HwMon
+			if config.HwMon.Index > 0 && controllerConfig.Index != config.HwMon.Index {
+				continue
+			}
+			if config.HwMon.Channel > 0 && controllerConfig.Channel != config.HwMon.Channel {
+				continue
+			}
+			config.HwMon.Index = controllerConfig.Index
```

Comment:
Just to be clear:
These two `if` statements are responsible for providing the "fallback", if a user still only has specified an index, right?

If so, is this in its current form guaranteeing that the association will stay the same?
In the assignments after the `if`'s, both the channel as well as the index will always be set, but the channel might not be the same as the index... right? Does this change break existing configurations because an index suddenly maps to a different channel? Or should this never happen because existing configurations will always map the index to the exact same channel since fan2go didn't even consider the channel before this change?

Reply:
I think you understand the change correctly. Fan2go enumerates fans the same way, so on a given system it will generate the same set of fans with the same index as it did before. So an existing config with an index will continue to work.

A config with a channel in it will match against the channel and not even look at the index. A config with both channel and index will be rejection by validation, but if it made its way to this code will match on the channel. A zero index or channel means "not set", which is consistent with fan2go before this patch for the index and consistent with the Linux hwmon subsystem for the channel.

If a system is _changed_ (fan added or removed), then the indices will likely change, but the channel numbers will stay consistent (unless you move a fan to a different connector, of course).

Reply:
Sounds good, thx!